File: Benchmarking/stats/euler52/euler52.py
import sys

limit = int(sys.argv[1])

def calc(limit):
  num = 0
  for i in range(5, limit+1):
    x1 = sorted(list(str(i)))
    x2 = sorted(list(str(i * 2)))
    x3 = sorted(list(str(i * 3)))
    x4 = sorted(list(str(i * 4)))
    x5 = sorted(list(str(i * 5)))
    x6 = sorted(list(str(i * 6)))
    sameDigits = False
    if x1 == x2 and x2 == x3 and x3 == x4 and x4 == x5 and x5 == x6:
      sameDigits = True
      num = i
  return num

print calc(limit)

File: Benchmarking/stats/euler52/euler52.java
import java.util.*;

public class euler52
{
	public static void main(String[] args)
	{
		int limit = Integer.parseInt(args[0]);
		calc(limit);
	}
	
	private static int calc(int limit)
	{
		int num = 0;
		for(int i = 5; i <= limit; i++)
		{
			String[] x1 = String.valueOf(i).split("");
			String[] x2 = String.valueOf(i * 2).split("");
			String[] x3 = String.valueOf(i * 3).split("");
			String[] x4 = String.valueOf(i * 4).split("");
			String[] x5 = String.valueOf(i * 5).split("");
			String[] x6 = String.valueOf(i * 6).split("");
			Arrays.sort(x1);
			Arrays.sort(x2);
			Arrays.sort(x3);
			Arrays.sort(x4);
			Arrays.sort(x5);
			Arrays.sort(x6);
			boolean sameDigits = false;
			if(Arrays.deepEquals(x1, x2) && Arrays.deepEquals(x2, x3) && Arrays.deepEquals(x3, x4) && Arrays.deepEquals(x4, x5) && Arrays.deepEquals(x5, x6))
			{
				sameDigits = true;
				num = i;
			}
		}
		return num;
	}
}
